exla/lib/exla.ex

```diff
@@ -355,6 +355,43 @@ defmodule EXLA do
     Nx.Defn.stream(function, args, Keyword.put(options, :compiler, EXLA))
   end
 
+  @doc """
+  Takes in a function, the templates variables and the compilation options
+  and returns the `EXLA.Executable` struct.
+
+  ## Examples
+
+      iex> fun = fn x, y -> Nx.add(Nx.sin(x), Nx.cos(y)) end
+      iex> args = [1.0, 2.0]
+      iex> module = EXLA.to_mlir_module(fun, args)
+      iex> EXLA.MLIR.Module.as_string(module)
+      ~c\"\"\"
```

Comment:
Btw. let's make `Module.as_string` return an actual string, not charlist :D

Reply:
Done